sunkit_image/stara.py

```diff
@@ -0,0 +1,154 @@
+from functools import partial
+
+import astropy.time as Time
+import astropy.units as u
+import matplotlib.pyplot as plt
+import numpy as np
+import sunpy.map
+from astropy.table import QTable
+from skimage.filters import median
+from skimage.measure import label, regionprops_table
+from skimage.morphology import disk, square, white_tophat
+from skimage.util import invert
+
+__all__ = ["stara", "get_regions", "plot_sunspots", "get_segs"]
+
+
+@u.quantity_input
+def stara(
+    smap,
+    circle_radius: u.deg = 100 * u.arcsec,
+    median_box: u.deg = 10 * u.arcsec,
+    threshold=6000,
+    limb_filter: u.percent = None,
+):
+    """
+    A method for automatically detecting sunspots in white-light data using
+    morphological operations.
+
+    Parameters
+    ----------
+    smap : `sunpy.map.GenericMap`
+        The map to apply the algorithm to.
+
+    circle_radius : `astropy.units.Quantity`, optional
+        The angular size of the structuring element used in the
+        `skimage.morphology.white_tophat`. This is the maximum radius of
+        detected features.
+
+    median_box : `astropy.units.Quantity`, optional
+        The size of the structuring element for the median filter, features
+        smaller than this will be averaged out.
+
+    threshold : `int`, optional
+        The threshold used for detection, this will be subject to detector
+        degradation. The default is a reasonable value for HMI continuum images.
+
+    limb_filter : `astropy.units.Quantity`, optional
+        If set, ignore features close to the limb within a percentage of the
+        radius of the disk. A value of 10% generally filters out false
+        detections around the limb with HMI continuum images.
+    """
```

Comment:
I think there should be a reference to the published papers on this method.

Reply:
Yes indeed. I have added references, please check and let me know if they are the correct way.